stan/math/prim/fun/lb_constrain.hpp

```diff
@@ -29,15 +32,42 @@ namespace math {
  * @param[in] lb lower bound on constrained output
  * @return lower bound constrained value corresponding to inputs
  */
-template <typename T, typename L>
+template <typename T, typename L, require_stan_scalar_t<T>* = nullptr>
 inline return_type_t<T, L> lb_constrain(const T& x, const L& lb) {
   using std::exp;
-  if (lb == NEGATIVE_INFTY) {
+  if (unlikely(is_negative_infinity(value_of(lb)))) {
```

Comment:
Yeah this looks wrong cause it's saying if any element has a lower bound of negative infinity, bail on them all.